src/utils/cosmosdb_model_versioned.ts

```diff
@@ -75,13 +76,16 @@ export const incVersion = (version: NonNegativeInteger) =>
 export abstract class CosmosdbModelVersioned<
   T,
   TN extends Readonly<T & Partial<NewVersionedModel>>,
-  TR extends Readonly<T & RetrievedVersionedModel>
-> extends CosmosdbModel<T, TN & BaseModel, TR> {
+  TR extends Readonly<T & RetrievedVersionedModel>,
+  IDENTITY extends keyof T,
+  PK extends keyof T | undefined = undefined
```

Comment:
I'd name this PartitionKey since PK might be "Primary Key" (which is not)

Reply:
ok